app/src/main/java/com/graddex/graddex/models/PokemonDetailsViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package com.graddex.graddex.models
+
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import okhttp3.*
+import java.io.IOException
+
+class PokemonDetailsViewModel : ViewModel() {
+
+    val pokemonSpriteFront: MutableLiveData<String> = MutableLiveData()
+    val pokemonSpriteBack: MutableLiveData<String> = MutableLiveData()
+    val pokemonName: MutableLiveData<String> = MutableLiveData()
+    val pokemonTypes: MutableLiveData<List<PokemonTypes>> = MutableLiveData()
+    val pokemonAbilities: MutableLiveData<List<PokemonAbilities>> = MutableLiveData()
+
+    fun syncPokemonDetails(url: String) {
+
+        val tag = "PokeAPI Details"
+
+        // Initialise rest client and implement a JSON adapter
+        val client = OkHttpClient()
+        val moshi = Moshi.Builder().add(KotlinJsonAdapterFactory()).build()
+        val adapter: JsonAdapter<PokemonDetailsResponse> = moshi.adapter(
+            PokemonDetailsResponse::class.java
+        )
+
```

Comment:
this might be good to move up as class properties, since they can be reused multiple times if `syncPokemonDetails` is called multiple times. just a slight efficiency thing